src/main/java/org/app/TransactionsHandler.java

```diff
@@ -3,36 +3,38 @@
 import java.math.BigDecimal;
 import java.math.RoundingMode;
 import java.time.ZonedDateTime;
-import java.util.concurrent.ConcurrentHashMap;
-import java.util.Map;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.Observable;
+import java.util.Observer;
 
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Service;
 
 import org.app.models.Statistics;
 import org.app.models.Transaction;
 
 @Service
-public class TransactionsHandler {
+public class TransactionsHandler implements Observer {
     private static TransactionsHandler instance;
-    private static ConcurrentHashMap<Integer,Transaction> storage;
-    private static int nextKey = 0;
+    private static LinkedList<Transaction> storage;
+    private static boolean statisticsCalculated = false;
```

Comment:
Cuando declaras variables de manera global, al no asignar un valor, la variable adquiere su valor default, que en caso de un `boolean` siempre será `false`

```suggestion
    private static boolean statisticsCalculated;
```